### PR TITLE
BUG: Fix droped result column in groupby with as_index False

### LINKED ISSUE #32240: NamedAgg with as_index= False drops aggregated values
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

file_name = "https://raw.githubusercontent.com/uiuc-cse/data-fa14/gh-pages/data/iris.csv"
df = pd.read_csv(file_name)

df[['species','petal_width']].groupby('species', as_index=False).agg(min_width=pd.NamedAgg(column='petal_width', aggfunc='min'))

```
#### Problem description
When I try to use `as_index=False`with named aggregations it seems to be dropping values and replacing them with index. This is the output of the above code.
|    | min_width   |
|---:|:------------|
|  0 | setosa      |
|  1 | versicolor  |
|  2 | virginica   |

#### Expected Output
|    | species    |   min_width |
|---:|:-----------|--------------:|
|  0 | setosa     |           0.1 |
|  1 | versicolor |           1   |
|  2 | virginica  |           1.4 |
#### Output of ``pd.show_versions()``

<details>

```
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.14.138+
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.1
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.1
setuptools       : 41.0.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.3.0
sqlalchemy       : 1.3.4
tables           : None
tabulate         : None
xarray           : 0.12.2
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None
```
</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #32240
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

The aggregate function *lost* the results of the aggregate operation, if the ```as_index=False``` flag was set and one of the result columns is relabeled. The index was reset at first, adding columns to the DataFrame. The relabeling operation only took the columns before resetting the index as input. So the weird error occurred. I changed the order of the index reset and the relabeling. This fixes the issue.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -963,16 +963,16 @@ def aggregate(self, func=None, *args, **kwargs):
...
-            self._insert_inaxis_grouper_inplace(result)
-            result.index = np.arange(len(result))
-
         if relabeling:
 
             # used reordered index of columns
             result = result.iloc[:, order]
             result.columns = columns
 
+        if not self.as_index:
```
Comment (mroeschke) [inline pandas/core/groupby/generic.py:972]: Could you put the issue number as a comment above this line and briefly describe the issue

**Comment by mroeschke** [comment]
One comment otherwise LGTM

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (phofl) [reply pandas/core/groupby/generic.py:977]: Hi, I have added the comment.

**Comment by phofl** [comment]
Done @mroeschke

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -963,16 +963,21 @@ def aggregate(self, func=None, *args, **kwargs):
...
-            self._insert_inaxis_grouper_inplace(result)
-            result.index = np.arange(len(result))
-
         if relabeling:
 
             # used reordered index of columns
             result = result.iloc[:, order]
             result.columns = columns
 
+        # GH 32240: The groupby function lost the result values if the as_index=False
```
Comment (jreback) [inline pandas/core/groupby/generic.py:972]: this comment is not needed

File: doc/source/whatsnew/v1.1.0.rst [hunk 3]
```diff
@@ -403,6 +403,7 @@ Groupby/resample/rolling
 - Bug in :meth:`GroupBy.apply` raises ``ValueError`` when the ``by`` axis is not sorted and has duplicates and the applied ``func`` does not mutate passed in objects (:issue:`30667`)
 - Bug in :meth:`DataFrameGroupby.transform` produces incorrect result with transformation functions (:issue:`30918`)
 - Bug in :meth:`DataFrame.groupby` and :meth:`Series.groupby` produces inconsistent type when aggregating Boolean series (:issue:`32894`)
+- Bug in :meth:`DataFrame.groupby` lost results, when ``as_index`` option was set to ``False``. The result values were replaced with the index values (:issue:`32240`).
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:406]: add when relabling the aggregation. Also this applies only to .agg, so use that reference ```:meth:`DataFrameGroupby.agg```

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.1.0.rst (hunk 3 above)
Comment (phofl) [reply doc/source/whatsnew/v1.1.0.rst:406]: Changed it

File: pandas/core/groupby/generic.py (hunk 2 above)
Comment (phofl) [reply pandas/core/groupby/generic.py]: Sorry, added it per request. I deleted the comment again.

**Comment by phofl** [comment]
Merge conflict in whats new.

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 4]
```diff
@@ -822,6 +822,41 @@ def test_groupby_aggregate_period_frame(func):
     tm.assert_frame_equal(result, expected)
 
 
+def test_grouby_agg_loses_results_with_as_index_false_relabel():
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_aggregate.py:825]: can you put these after the other relablng tests (IIRC in this same file)

File: doc/source/whatsnew/v1.1.0.rst [hunk 5]
```diff
@@ -426,6 +426,7 @@ Groupby/resample/rolling
 - Bug in :meth:`DataFrameGroupby.transform` produces incorrect result with transformation functions (:issue:`30918`)
 - Bug in :meth:`DataFrame.groupby` and :meth:`Series.groupby` produces inconsistent type when aggregating Boolean series (:issue:`32894`)
 - Bug in :meth:`DataFrame.resample` where an ``AmbiguousTimeError`` would be raised when the resulting timezone aware :class:`DatetimeIndex` had a DST transition at midnight (:issue:`25758`)
+- Bug in :meth:`DataFrameGroupby.agg` lost results, when ``as_index`` option was set to ``False`` and the result columns were relabeled. The result values were replaced with the index values (:issue:`32240`).
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:429]: can you add a section with previous / current behavior (you can do in api breaking changes, its not really an api change, but a substantially change).

**Review by jreback — CHANGES_REQUESTED** [review]
looks good a couple of comments

File: doc/source/whatsnew/v1.1.0.rst (hunk 5 above)
Comment (phofl) [reply doc/source/whatsnew/v1.1.0.rst:464]: Done. Hope I inserted it in the right section

File: pandas/tests/groupby/aggregate/test_aggregate.py (hunk 4 above)
Comment (phofl) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: Done

**Comment by mroeschke** [comment]
Looks good. Looks like there are some failing code checks

**Comment by phofl** [comment]
> 
> 
> Looks good. Looks like there are some failing code checks

Missed that one, is green now.

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
let's rebase this on top of #34012 (or could wait, likey to merge that one tomorrow). this migth be fine but will want to re-look.

**Comment by phofl** [comment]
@jreback Merged master after the pull request was merged. Seems to be fine.

**Comment by phofl** [comment]
@jreback  merged master

File: doc/source/whatsnew/v1.1.0.rst [hunk 6]
```diff
@@ -715,6 +715,43 @@ apply and applymap on ``DataFrame`` evaluates first row/column only once
     df.apply(func, axis=1)
 
 
+.. _whatsnew_110.api_breaking.groupby_results_los_as_index_false:
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:718]: typo here (los -> lost)?

File: doc/source/whatsnew/v1.1.0.rst [hunk 7]
```diff
@@ -715,6 +715,43 @@ apply and applymap on ``DataFrame`` evaluates first row/column only once
...
+
+:meth:`DataFrameGroupby.agg` lost results with ``as_index`` ``False`` when relabeling columns
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:720]: can you move this right under the other section that deals with `as_index=False`

File: doc/source/whatsnew/v1.1.0.rst (hunk 5 above)
Comment (jreback) [reply doc/source/whatsnew/v1.1.0.rst:1032]: you don't also need this

**Review by jreback — CHANGES_REQUESTED** [review]
some doc comments, looks good otherwise.

File: doc/source/whatsnew/v1.1.0.rst (hunk 7 above)
Comment (phofl) [reply doc/source/whatsnew/v1.1.0.rst]: Done

**Comment by phofl** [comment]
@jreback Changed the docs, test failure seems unrelated

**Comment by TomAugspurger** [comment]
Will need a similar fix as https://github.com/pandas-dev/pandas/pull/34721/files#diff-fb957a9fbb8d57904092e2bbe06b1b2bR99-R109.

**Comment by jreback** [comment]
looks good, can you rebase to fix the conflict (the other fail is ok), ping when ready.

**Comment by phofl** [comment]
@jreback fixed conflict, failure unrelated. Can be merged

**Comment by jreback** [comment]
can you merge master and ping when green

**Comment by jreback** [comment]
@phofl if you can merge master will have another look

**Comment by phofl** [comment]
Sorry, was a Bit busy the last few days. Will merge master tomorrow and ping when ready

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**